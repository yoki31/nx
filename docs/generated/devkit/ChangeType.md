# Enumeration: ChangeType

## Table of contents

### Enumeration Members

- [Delete](../../devkit/documents/ChangeType#delete)
- [Insert](../../devkit/documents/ChangeType#insert)

## Enumeration Members

### Delete

• **Delete** = `"Delete"`

---

### Insert

• **Insert** = `"Insert"`
