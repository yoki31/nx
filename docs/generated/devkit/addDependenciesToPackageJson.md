# Function: addDependenciesToPackageJson

â–¸ **addDependenciesToPackageJson**(`tree`, `dependencies`, `devDependencies`, `packageJsonPath?`, `keepExistingVersions?`): [`GeneratorCallback`](../../devkit/documents/GeneratorCallback)

Add Dependencies and Dev Dependencies to package.json

For example:

```typescript
addDependenciesToPackageJson(tree, { react: 'latest' }, { jest: 'latest' });
```

This will **add** `react` and `jest` to the dependencies and devDependencies sections of package.json respectively.

#### Parameters

| Name                    | Type                                  | Description                                                                 |
| :---------------------- | :------------------------------------ | :-------------------------------------------------------------------------- |
| `tree`                  | [`Tree`](../../devkit/documents/Tree) | Tree representing file system to modify                                     |
| `dependencies`          | `Record`\<`string`, `string`\>        | Dependencies to be added to the dependencies section of package.json        |
| `devDependencies`       | `Record`\<`string`, `string`\>        | Dependencies to be added to the devDependencies section of package.json     |
| `packageJsonPath?`      | `string`                              | Path to package.json                                                        |
| `keepExistingVersions?` | `boolean`                             | If true, prevents existing dependencies from being bumped to newer versions |

#### Returns

[`GeneratorCallback`](../../devkit/documents/GeneratorCallback)

Callback to install dependencies only if necessary, no-op otherwise
