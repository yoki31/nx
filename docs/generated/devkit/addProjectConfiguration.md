# Function: addProjectConfiguration

â–¸ **addProjectConfiguration**(`tree`, `projectName`, `projectConfiguration`, `standalone?`): `void`

Adds project configuration to the Nx workspace.

#### Parameters

| Name                   | Type                                                                  | Default value | Description                                                             |
| :--------------------- | :-------------------------------------------------------------------- | :------------ | :---------------------------------------------------------------------- |
| `tree`                 | [`Tree`](../../devkit/documents/Tree)                                 | `undefined`   | the file system tree                                                    |
| `projectName`          | `string`                                                              | `undefined`   | unique name. Often directories are part of the name (e.g., mydir-mylib) |
| `projectConfiguration` | [`ProjectConfiguration`](../../devkit/documents/ProjectConfiguration) | `undefined`   | project configuration                                                   |
| `standalone`           | `boolean`                                                             | `true`        | whether the project is configured in workspace.json or not              |

#### Returns

`void`
