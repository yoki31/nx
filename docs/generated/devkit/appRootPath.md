# Variable: appRootPath

â€¢ `Const` **appRootPath**: `string` = `workspaceRoot`

The root of the workspace.

**`Deprecated`**

use workspaceRoot instead
