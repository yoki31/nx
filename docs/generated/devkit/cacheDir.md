# Variable: cacheDir

â€¢ `Const` **cacheDir**: `string`

Path to the directory where Nx stores its cache and daemon-related files.
