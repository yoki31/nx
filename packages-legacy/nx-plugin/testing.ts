export * from '@nx/plugin/testing';
