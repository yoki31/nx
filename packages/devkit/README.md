<p style="text-align: center;">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-dark.svg">
    <img alt="Nx - Smart Monorepos · Fast CI" src="https://raw.githubusercontent.com/nrwl/nx/master/images/nx-light.svg" width="100%">
  </picture>
</p>

{{links}}

<hr>

# Nx: Smart Monorepos · Fast CI

Nx is a build system with built-in tooling and advanced CI capabilities. It helps you maintain and scale monorepos, both locally and on CI.

This package contains a set of utilities for creating Nx plugins.

{{content}}
