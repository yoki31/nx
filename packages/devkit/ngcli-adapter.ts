/**
 * @category Ng CLI Adapter
 */
// eslint-disable-next-line @typescript-eslint/no-restricted-imports
export {
  wrapAngularDevkitSchematic,
  mockSchematicsForTesting,
  NxScopedHost,
} from 'nx/src/adapter/ngcli-adapter';
