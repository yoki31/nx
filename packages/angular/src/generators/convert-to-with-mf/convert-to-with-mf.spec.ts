import { addProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import {
  STANDARD_HOST_MFE_CONFIG,
  STANDARD_REMOTE_MFE_CONFIG,
  OLD_OBJECT_SHARED_SYNTAX,
  ERROR_NAME_DOESNT_MATCH,
  ERROR_SHARED_PACKAGES_DOESNT_MATCH,
} from './convert-to-with-mf.test-data';
import convertToWithMF from './convert-to-with-mf';

describe('convertToWithMF', () => {
  it('should migrate a standard previous generated host config correctly', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'host1', {
      name: 'host1',
      root: 'apps/host1',
      sourceRoot: 'apps/host1/src',
      targets: {
        build: {
          executor: '@nx/angular:webpack-browser',
          options: {
            customWebpackConfig: {
              path: 'apps/host1/webpack.config.js',
            },
          },
        },
      },
    });

    tree.write('apps/host1/webpack.config.js', STANDARD_HOST_MFE_CONFIG);

    // ACT
    await convertToWithMF(tree, {
      project: 'host1',
    });

    // ASSERT
    const webpackSource = tree.read('apps/host1/webpack.config.js', 'utf-8');
    expect(webpackSource).toMatchSnapshot();
  });

  it('should migrate a standard previous generated remote config correctly', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'remote1', {
      name: 'remote1',
      root: 'apps/remote1',
      sourceRoot: 'apps/remote1/src',
      targets: {
        build: {
          executor: '@nx/angular:webpack-browser',
          options: {
            customWebpackConfig: {
              path: 'apps/remote1/webpack.config.js',
            },
          },
        },
      },
    });

    tree.write('apps/remote1/webpack.config.js', STANDARD_REMOTE_MFE_CONFIG);

    // ACT
    await convertToWithMF(tree, {
      project: 'remote1',
      skipFormat: true,
    });

    // ASSERT
    const webpackSource = tree.read('apps/remote1/webpack.config.js', 'utf-8');
    expect(webpackSource).toMatchSnapshot();
  });

  it('should migrate a standard previous generated remote config using object shared syntax correctly', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'remote1', {
      name: 'remote1',
      root: 'apps/remote1',
      sourceRoot: 'apps/remote1/src',
      targets: {
        build: {
          executor: '@nx/angular:webpack-browser',
          options: {
            customWebpackConfig: {
              path: 'apps/remote1/webpack.config.js',
            },
          },
        },
      },
    });

    tree.write('apps/remote1/webpack.config.js', OLD_OBJECT_SHARED_SYNTAX);

    // ACT
    await convertToWithMF(tree, {
      project: 'remote1',
      skipFormat: true,
    });

    // ASSERT
    const webpackSource = tree.read('apps/remote1/webpack.config.js', 'utf-8');
    expect(webpackSource).toMatchSnapshot();
  });

  it('should throw when the uniqueName doesnt match the project name', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'remote1', {
      name: 'remote1',
      root: 'apps/remote1',
      sourceRoot: 'apps/remote1/src',
      targets: {
        build: {
          executor: '@nx/angular:webpack-browser',
          options: {
            customWebpackConfig: {
              path: 'apps/remote1/webpack.config.js',
            },
          },
        },
      },
    });

    tree.write('apps/remote1/webpack.config.js', ERROR_NAME_DOESNT_MATCH);

    // ACT & ASSERT
    await expect(
      convertToWithMF(tree, {
        project: 'remote1',
      })
    ).rejects.toThrow();
  });

  it('should throw when the shared npm packages configs has been modified', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'host1', {
      name: 'host1',
      root: 'apps/host1',
      sourceRoot: 'apps/host1/src',
      targets: {
        build: {
          executor: '@nx/angular:webpack-browser',
          options: {
            customWebpackConfig: {
              path: 'apps/host1/webpack.config.js',
            },
          },
        },
      },
    });

    tree.write(
      'apps/host1/webpack.config.js',
      ERROR_SHARED_PACKAGES_DOESNT_MATCH
    );

    // ACT & ASSERT
    await expect(
      convertToWithMF(tree, {
        project: 'host1',
      })
    ).rejects.toThrow();
  });
});
