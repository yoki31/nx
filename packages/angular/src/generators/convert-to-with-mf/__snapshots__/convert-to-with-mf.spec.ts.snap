// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToWithMF should migrate a standard previous generated host config correctly 1`] = `
"const { withModuleFederation } = require('@nx/angular/module-federation');
const config = require('./module-federation.config');
module.exports = withModuleFederation(config);
"
`;

exports[`convertToWithMF should migrate a standard previous generated remote config correctly 1`] = `
"const { withModuleFederation } = require('@nx/angular/module-federation');
const config = require('./module-federation.config');
module.exports = withModuleFederation(config);
"
`;

exports[`convertToWithMF should migrate a standard previous generated remote config using object shared syntax correctly 1`] = `
"const { withModuleFederation } = require('@nx/angular/module-federation');
const config = require('./module-federation.config');
module.exports = withModuleFederation(config);
"
`;
